jython-processors/trigonometric-transformation/processor.py

```diff
@@ -1,20 +1,117 @@
-def process(row):
-    import math
-    import json
-    periodical_column_name = params.get('periodical_column')
-    periodical_column_value = row[periodical_column_name]
-    column_period = params.get('column_period')
-    arg_val = math.pi/2.0-(math.pi*int(periodical_column_value))/(column_period/2.0)
+#Packages import 
+import math
+
+#Class definition :
+class TrigonometricTransformer():
+    """
+    The TrigonometricTransformer's goal is to set up all the functions
+    that will be called by the recipe's processor. 
+    These functions are defined as lambdas functions contextualized by the end user's selection. 
+    This way the condition in order to compute the asked functions is checked only once.
+    
+    - 'build_trigonometric_transformer' : is a method allowing to setup the lambdas functions 
+        called by the TrigonometricTransformer() class.
+    - 'compute_trigonometric_transform' : is a method allowing to map the dataset variable 
+        to the functions set by the build_trigonometric_transformer() method.
+    - 'compute_argument' : is a methiod allowing to convert a numerical value from a 
+        periodical feature  into a trigonometric argument. 
+    - 'compute_trigonometric_features' : is a method allowing, if 'process_as_time_dimension==True', to 
```

Comment:
What is the % time saving benefit of pre-computation? Can you benchmark it compared to a simple approach?